Mage.Sets/src/mage/cards/g/GhoulishImpetus.java

```diff
@@ -0,0 +1,67 @@
+package mage.cards.g;
+
+import mage.abilities.Ability;
+import mage.abilities.common.DiesAttachedTriggeredAbility;
+import mage.abilities.common.SimpleStaticAbility;
+import mage.abilities.common.delayed.AtTheBeginOfNextEndStepDelayedTriggeredAbility;
+import mage.abilities.effects.common.AttachEffect;
+import mage.abilities.effects.common.CreateDelayedTriggeredAbilityEffect;
+import mage.abilities.effects.common.ReturnToBattlefieldUnderYourControlTargetEffect;
+import mage.abilities.effects.common.combat.GoadAttachedEffect;
+import mage.abilities.effects.common.continuous.BoostEnchantedEffect;
+import mage.abilities.effects.common.continuous.GainAbilityAttachedEffect;
+import mage.abilities.keyword.DeathtouchAbility;
+import mage.abilities.keyword.EnchantAbility;
+import mage.cards.CardImpl;
+import mage.cards.CardSetInfo;
+import mage.constants.*;
+import mage.target.TargetPermanent;
+import mage.target.common.TargetCreaturePermanent;
+
+import java.util.UUID;
+
+/**
+ * @author Susucr
+ */
+public final class GhoulishImpetus extends CardImpl {
+
+    public GhoulishImpetus(UUID ownerId, CardSetInfo setInfo) {
+        super(ownerId, setInfo, new CardType[]{CardType.ENCHANTMENT}, "{2}{B}");
+
+        this.subtype.add(SubType.AURA);
+
+        // Enchant creature
+        TargetPermanent auraTarget = new TargetCreaturePermanent();
+        this.getSpellAbility().addTarget(auraTarget);
+        this.getSpellAbility().addEffect(new AttachEffect(Outcome.BoostCreature));
+        this.addAbility(new EnchantAbility(auraTarget));
+
+        // Enchanted creature gets +1/+1, has deathtouch, and is goaded.
+        Ability ability = new SimpleStaticAbility(new BoostEnchantedEffect(1, 1));
+        ability.addEffect(new GainAbilityAttachedEffect(DeathtouchAbility.getInstance(), AttachmentType.AURA).concatBy(", "));
+        ability.addEffect(new GoadAttachedEffect().concatBy(", and"));
+        this.addAbility(ability);
+
+        // When enchanted creature dies, return Ghoulish Impetus to the battlefield at the beginning of the next end step.
+        this.addAbility(new DiesAttachedTriggeredAbility(
+            new CreateDelayedTriggeredAbilityEffect(
+                new AtTheBeginOfNextEndStepDelayedTriggeredAbility(
+                    new ReturnToBattlefieldUnderYourControlTargetEffect()
+                        .setText("return {this} to the battlefield"),
+                    TargetController.ANY
+                )
+            ).setText("return {this} to the battlefield at the beginning of the next end step"),
```

Comment:
I think you also have to `setTriggerPhrase("")`

Reply:
Don't think it is needed?

Here is current card rule (with the "and and goaded", that one you are correct), the trigger on death, and the delayed trigger at end of turn:
![andand](https://github.com/magefree/mage/assets/34709007/5e9f6797-d896-40b2-840b-d9dd39f54846)
![dying trigger](https://github.com/magefree/mage/assets/34709007/e8329db2-60ce-4d46-a5dc-0c1e0fc139c4)
![returntrigger](https://github.com/magefree/mage/assets/34709007/7a862f7e-add1-490c-be20-edad68201940)

Reply:
Huh okay, guess that works, I've needed that in the past for some reason but can't remember the details now.